grade-management-new/GradeManagement.Client/Pages/Courses.razor

```diff
@@ -0,0 +1,137 @@
+﻿@page "/courses"
+
+@implements IDisposable
+@inject HttpClient Http
+@inject IDialogService DialogService
+@inject CrudSnackbarService SnackbarService
+@inject SubjectService SubjectService
+@inject NavigationManager NavigationManager
+
+
+<LoadingComponent @ref="loadingComponentRef" LongTask="@_loading">
+    <MudDataGrid T="Course" Items="_courses" EditMode="DataGridEditMode.Form" ReadOnly="false" CommittedItemChanges="@CommittedItemChanges" RowClick="@RowClickEvent">
+        <Columns>
+            <PropertyColumn Property="x => x.Id" Title="Id" IsEditable="false"/>
+            <PropertyColumn Property="x => x.Name" Title="Name"/>
+            <PropertyColumn Property="x => x.MoodleCourseId" Title="Moodle Course Id"/>
+            <TemplateColumn Title="Semester">
+                <CellTemplate>
+                    @context.Item.Semester.Name
+                </CellTemplate>
+                <EditTemplate>
+                    <SelectSemester @bind-Value="@context.Item.Semester"/>
+                </EditTemplate>
+            </TemplateColumn>
+            <TemplateColumn Title="Language">
+                <CellTemplate>
+                    @context.Item.Language.Name
+                </CellTemplate>
+                <EditTemplate>
+                    <SelectLanguage @bind-Value="@context.Item.Language"/>
+                </EditTemplate>
+            </TemplateColumn>
+            <DataGridActions TItem="Course" OnEditItem="@StartEditing" OnDeleteItem="@DeleteItem"/>
+        </Columns>
+    </MudDataGrid>
+    <MudFab Style="position:fixed; bottom:50px;right:50px" Color="@Color.Primary" Icon="@Icons.Material.Filled.Add" OnClick="@OpenDialog"/>
+</LoadingComponent>
+
+@code {
+    private List<Course> _courses = new();
+    private LoadingComponent? loadingComponentRef;
+    private bool _isLoading = true;
+
+    readonly string BaseUrl = Endpoints.COURSES;
+
+    private void RowClickEvent(DataGridRowClickEventArgs<Course> args)
+    {
+        var courseId = args.Item.Id;
+        NavigationManager.NavigateTo($"/courses/{courseId}");
+    }
+
+    private async Task _loading()
+    {
+        _courses = await SubjectService.LoadCourses();
+    }
+
+
+    private async Task OpenDialog()
+    {
+        var dialog = await DialogService.ShowAsync<NewCourseDialog>("Hello");
+        var result = await dialog.Result;
+        if (!result.Canceled && result.Data != null)
+        {
+            var newCourse = result.Data as Course;
+            newCourse.SubjectId = SubjectService.CurrentSubject.Id;
+            _courses.Add(newCourse);
+            var response = await Http.PostAsJsonAsync(BaseUrl, newCourse);
+            if (response.IsSuccessStatusCode)
+            {
+                Course newObject = await response.Content.ReadFromJsonAsync<Course>();
```

Comment:
az is bevett szokás szokott lenni, hogy nem a választ használjuk fel a lista frissítésére, hanem befrissítjük a teljes listát, mint amikor ide navigáltunk